protocols/v2/codec-sv2/src/decoder.rs

```diff
@@ -1,3 +1,43 @@
+//! # Stratum V2 Codec Decoder
+//!
+//! This module provides functionality for decoding Stratum V2 messages, including support for
+//! the Noise protocol for secure communication.
+//!
+//! ## Features
+//!
+//! * **Standard Decoder**: Decodes Stratum V2 frames without encryption.
+//! * **Noise Decoder**: Decodes Stratum V2 frames with Noise protocol encryption.
+//!
+//! ## Types
+//!
+//! * `WithoutNoise`: Decoder for Sv2 frames without Noise protocol support.
+//! * `WithNoise`: Decoder for Sv2 frames with Noise protocol support (requires the `noise_sv2`
+//!   feature).
+//! * `StandardEitherFrame`: Represents an encoded or decoded frame that could be either a regular
+//!   or Noise-protected frame.
+//! * `StandardSv2Frame`: Represents an encoded or decoded Stratum V2 frame.
+//! * `StandardNoiseDecoder`: Decoder for Stratum V2 frames with Noise protocol support.
+//! * `StandardDecoder`: Decoder for Stratum V2 frames without Noise protocol support.
+//!
+//! ## Usage
+//!
+//! This module is designed to be used to decode incoming Sv2 messages, with optional Noise
+//! protocol encryption support for secure communication.
+//!
+//! ## Example
+//!
+//! ```ignore
+//! use codec_sv2::decoder::{StandardDecoder, StandardNoiseDecoder};
+//!
+//! // Create a standard decoder
+//! let mut decoder: StandardDecoder<MyFrameType> = StandardDecoder::new();
+//!
+//! // Create a noise decoder (requires the `noise_sv2` feature)
+//! #[cfg(feature = "noise_sv2")]
+//! let mut noise_decoder: StandardNoiseDecoder<MyFrameType> = StandardNoiseDecoder::new();
+//! ```
+//!
+
```

Comment:
this header is not going to be displayed in Rust Docs, because the `decoder` module is not exposed

overall I'd recommend to regularly do `cargo doc --all-features --open` to render the docs so you can have a good idea of what the end-user is going to see

Reply:
I think that we can change how we export thing in order to include it without changing the Api

Reply:
sorry @Fi3 I don't fully understand, can you elaborate?

Reply:
We can add the `--document-private-items` to expose documentation for private modules like:

```sh
cargo doc --document-private-items --all-features --open
```

I left a [comment here](https://github.com/stratum-mining/stratum/issues/1130#issuecomment-2347055483) with some thoughts.

Essentially, I am not sure if you can publish the private crate documentation to crates.io, or if we would want to do that. BUT, we can use it to guide us and view our docs on private modules while we develop the documentation.

Either way, my example here sucks 😝.

Reply:
yep or we can make the module public in this case: we make codec_sv2::decoder public and in the module we make private all the things that are not exported by lib.rs so we don't change the api.

Reply:
I agree, we should make these modules public.

Reply:
Agree. Added as TODO in #1129.